src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -24,9 +24,15 @@ internal sealed class EnumConverter<T> : JsonConverter<T>
 
         private readonly JsonNamingPolicy? _namingPolicy;
 
-        private readonly ConcurrentDictionary<ulong, JsonEncodedText> _nameCache;
+        /// <summary>
+        /// Holds a mapping from enum value to text that might be formatted with <see cref="_namingPolicy" />.
+        /// </summary>
+        private readonly ConcurrentDictionary<ulong, JsonEncodedText> _nameCacheForWriting;
 
-        private ConcurrentDictionary<ulong, JsonEncodedText>? _dictionaryKeyPolicyCache;
+        /// <summary>
+        /// Holds a mapping from text that might be formatted with <see cref="_namingPolicy" /> to enum value.
+        /// </summary>
+        private ConcurrentDictionary<string, T>? _nameCacheForReading;
```

Comment:
IIRC this is only set by the ctor so can also be marked readonly.
```suggestion
        private readonly ConcurrentDictionary<string, T>? _nameCacheForReading;
```